services/tests/literpc_tpu_quic_server_integrationtest.rs

```diff
@@ -0,0 +1,444 @@
+    max_number_of_connections: 8,
+    unistream_timeout: Duration::from_millis(500),
+    write_timeout: Duration::from_secs(1),
+    number_of_transactions_per_unistream: 10,
+};
+
+async fn start_literpc_client(
+    streamer_listen_addrs: SocketAddr,
+    literpc_validator_identity: Arc<Keypair>,
+) -> anyhow::Result<()> {
+    info!("Start lite-rpc test client ...");
+
+    let fanout_slots = 4;
+
+    // (String, Vec<u8>) (signature, transaction)
+    let (sender, _) = tokio::sync::broadcast::channel(MAXIMUM_TRANSACTIONS_IN_QUEUE);
+    let broadcast_sender = Arc::new(sender);
+    let (certificate, key) = new_self_signed_tls_certificate(
+        literpc_validator_identity.as_ref(),
+        IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+    )
+    .expect("Failed to initialize QUIC connection certificates");
+
+    let tpu_connection_manager =
+        TpuConnectionManager::new(certificate, key, fanout_slots as usize).await;
+
+    // this effectively controls how many connections we will have
+    let mut connections_to_keep: HashMap<Pubkey, SocketAddr> = HashMap::new();
+    let addr1 = UdpSocket::bind("127.0.0.1:0")
+        .unwrap()
+        .local_addr()
+        .unwrap();
+    connections_to_keep.insert(
+        Pubkey::from_str("1111111jepwNWbYG87sgwnBbUJnQHrPiUJzMpqJXZ")?,
+        addr1,
+    );
+
+    let addr2 = UdpSocket::bind("127.0.0.1:0")
+        .unwrap()
+        .local_addr()
+        .unwrap();
+    connections_to_keep.insert(
+        Pubkey::from_str("1111111k4AYMctpyJakWNvGcte6tR8BLyZw54R8qu")?,
+        addr2,
+    );
+
+    // this is the real streamer
+    connections_to_keep.insert(literpc_validator_identity.pubkey(), streamer_listen_addrs);
+
+    // get information about the optional validator identity stake
+    // populated from get_stakes_for_identity()
+    let identity_stakes = IdentityStakes {
+        peer_type: ConnectionPeerType::Staked,
+        stakes: if STAKE_CONNECTION { 30 } else { 0 }, // stake of lite-rpc
+        min_stakes: 0,
+        max_stakes: 40,
+        total_stakes: 100,
+    };
+
+    // solana_streamer::nonblocking::quic: Peer type: Staked, stake 30, total stake 0, max streams 128 receive_window Ok(12320) from peer 127.0.0.1:8000
+
+    tpu_connection_manager
+        .update_connections(
+            broadcast_sender.clone(),
+            connections_to_keep,
+            identity_stakes,
+            // note: tx_store is useless in this scenario as it is never changed; it's only used to check for duplicates
+            empty_tx_store().clone(),
+            QUIC_CONNECTION_PARAMS,
+        )
+        .await;
+
+    // TODO this is a race
+    sleep(Duration::from_millis(1500)).await;
+
+    for i in 0..SAMPLE_TX_COUNT {
+        let raw_sample_tx = build_raw_sample_tx(i);
+        debug!(
+            "broadcast transaction {} to {} receivers: {}",
+            raw_sample_tx.0,
+            broadcast_sender.receiver_count(),
+            format!("hi {}", i)
```

Comment:
format message correctly.

Reply:
will remove the debug. it's quite worthless and confusing